fastdom.js

```diff
@@ -120,28 +124,51 @@ function scheduleFlush(fastdom) {
 }
 
 /**
- * Runs queued `read` and `write` tasks.
+ * Schedules a new read
+ * batch if one isn't pending.
+ *
+ * @private
+ */
+
+function scheduleReadFlush(fastdom) {
+  if (fastdom.scheduledRead) return;
+  fastdom.scheduledRead = true;
+  if(fastdom._isWriting){
+    setTimeout(fastdom._flushReads, 0);
+  } else {
+    fastdom.immediate.then(fastdom._flushReads);
+    //in case we use setTimeout or a bad Promise polyfill, setTimeout can be called later than a rAF.
+    //by additionally using the ordinary scheduleFlush function we make sure that reads are called as early as possible.
+    scheduleFlush(fastdom);
```

Comment:
There are indeed inconsistencies cross-platform with browser with micro tasks running and those who don't. I think these changes are neglectable though. The order of read tasks, write tasks and frames are the same in any browser.

On the other hand, we do this just for backwards compatibility and to be always. reads which aren't nested in a write can always be called sync.

It's your choice.